pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -227,6 +227,12 @@ static PyObject *get_values(PyObject *obj) {
     PyObject *values = PyObject_GetAttrString(obj, "values");
     PRINTMARK();
 
```

Comment:
I'll add comments here and where I define it.

I don't think it should be part of the interface.

FWIW, other EA don't have any problem serializing.

Reply:
Yeah I don't really understand what's going on. Presumably something to do with DatetimeArray not being recognized as a datetlike, so they values date format isn't being applied to its values?

Reply:
hmm sounds like a more general  issue. I agree that the ``.to_json()`` needs to coerce, just see below at like 238, maybe just change this to ``.to_numpy()`` and it will just work. This is really smelly and should not be here (``_to_json_values``).